plugin.php

```diff
@@ -35,21 +29,41 @@ function check_for_redirect( $args ) {
 
 function redirect_to_advert( $url, $code ) {
   if ( doAdvert ) {
-	$redirectUrl = getRedirect();
-    if ( redirectService == 'f' ) { // Use adfocus
-      return ADFOCUS_DOMAIN . '/serve/sitelinks/?id=' . ADFOCUS_ID . '&url=' . $redirectUrl;
-    } else if ( redirectService == 'a' ) { // Adfly
-	  return ADFLY_DOMAIN . '/' . ADFLY_ID . '/' . $redirectUrl;
-    } else if ( redirectService == 'o' ) { // OUO.io
-      return OUO_DOMAIN . '/qs/' . OUO_ID . '?s=' . $redirectUrl;
+    $redirectUrl = getRedirect();
+    switch ( redirectService ) { 
+      case 'f': // Use adfocus
+        return ADFOCUS_DOMAIN . '/serve/sitelinks/?id=' . ADFOCUS_ID . '&url=' . $redirectUrl;
+      case 'a': // Adfly
+        return ADFLY_DOMAIN . '/' . ADFLY_ID . '/' . $redirectUrl;
+      case 'o': // OUO.io
+        return OUO_DOMAIN . '/qs/' . OUO_ID . '?s=' . $redirectUrl;
+      case 'l': // linkvertise.com
+        return getLinkvertise(LINKVERTISE_ID,$redirectUrl);
     }
   }
   return $url; // If none of those redirect services, forward to the normal URL
 }
 
 function getRedirect(){
-	$actual_link = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]"; // get the current url
-	$pieces = explode('/', $actual_link); // split the url into an arrray seperated by /
-	$last_word = array_pop($pieces); //  get the keyword - this may not work if you use a plugin to allow slashes in your shortened url
-	return '//' . $_SERVER['SERVER_NAME'] . '/' . $last_word; // replace the '/' after $_SERVER['SERVER_NAME' if your yourls is not in your base domain, such as '/shorten/'
+  $protocol = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") ;
+  $actual_link = $protocol . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]"; // get the current url
+  $pieces = explode('/', $actual_link); // split the url into an arrray seperated by /
+  $last_word = array_pop($pieces); //  get the keyword - this may not work if you use a plugin to allow slashes in your shortened url
+  return $protocol . '://' . $_SERVER['SERVER_NAME'] . '/' . $last_word; // replace the '/' after $_SERVER['SERVER_NAME' if your yourls is not in your base domain, such as '/shorten/'
+}
+// About Linkvertise
+function btoa($str) {
```

Comment:
Could we use the default [`base64_encode()`](https://www.php.net/manual/en/function.base64-encode.php) function instead?

Reply:
I can try to test it

Reply:
I meant moreso why do we need the buffer conversion here? Can we not just use `base64_encode` directly in `getLinkvertise`?  I was under the impression `btoa` in JavaScript functions the same as `base64_encode` in PHP, does it not?

Reply:
Okay, I'll try it, but I'm not sure if it will still pass the verification.